daemon/api_users.go

```diff
@@ -341,196 +313,34 @@ func createUser(c *Command, createData postUserCreateData) Response {
-		// return a single userResponseData in this case
-		return SyncResponse(&result)
 	} else {
-		return SyncResponse([]userResponseData{result})
-	}
-}
-
-func getUserDetailsFromStore(theStore snapstate.StoreService, email string) (string, *osutil.AddUserOptions, error) {
-	v, err := theStore.UserInfo(email)
-	if err != nil {
-		return "", nil, fmt.Errorf("cannot create user %q: %s", email, err)
-	}
-	if len(v.SSHKeys) == 0 {
-		return "", nil, fmt.Errorf("cannot create user for %q: no ssh keys found", email)
-	}
-
-	// Amend information where the key came from to ensure it can
-	// be update/replaced later
-	for i, k := range v.SSHKeys {
-		v.SSHKeys[i] = fmt.Sprintf(`%s # snapd {"origin":"store","email":%q}`, k, email)
-	}
-
-	gecos := fmt.Sprintf("%s,%s", email, v.OpenIDIdentifier)
-	opts := &osutil.AddUserOptions{
-		SSHKeys: v.SSHKeys,
-		Gecos:   gecos,
-	}
-	return v.Username, opts, nil
-}
-
-func createAllKnownSystemUsers(st *state.State, modelAs *asserts.Model, serialAs *asserts.Serial, createData *postUserCreateData) Response {
-	var createdUsers []userResponseData
-	headers := map[string]string{
-		"brand-id": modelAs.BrandID(),
-	}
-
-	st.Lock()
-	db := assertstate.DB(st)
-	assertions, err := db.FindMany(asserts.SystemUserType, headers)
-	st.Unlock()
-	if err != nil && !asserts.IsNotFound(err) {
-		return BadRequest("cannot find system-user assertion: %s", err)
-	}
-
-	for _, as := range assertions {
-		email := as.(*asserts.SystemUser).Email()
-		// we need to use getUserDetailsFromAssertion as this verifies
-		// the assertion against the current brand/model/time
-		username, opts, err := getUserDetailsFromAssertion(st, modelAs, serialAs, email)
-		if err != nil {
-			logger.Noticef("ignoring system-user assertion for %q: %s", email, err)
-			continue
-		}
-		// ignore already existing users
-		if _, err := userLookup(username); err == nil {
-			continue
-		}
-
-		// FIXME: duplicated code
-		opts.Sudoer = createData.Sudoer
-		opts.ExtraUsers = !release.OnClassic
-
-		if err := osutilAddUser(username, opts); err != nil {
-			return InternalError("cannot add user %q: %s", username, err)
+		for _, cu := range createdUsers {
+			createdUsersResponse = append(createdUsersResponse, userResponseData{
+				Username: cu.Username,
+				SSHKeys:  cu.SSHKeys,
+			})
 		}
-		if err := setupLocalUser(st, username, email); err != nil {
-			return InternalError("%s", err)
-		}
-		createdUsers = append(createdUsers, userResponseData{
-			Username: username,
-			SSHKeys:  opts.SSHKeys,
-		})
 	}
-
-	return SyncResponse(createdUsers)
+	return SyncResponse(createdUsersResponse)
 }
 
-func getUserDetailsFromAssertion(st *state.State, modelAs *asserts.Model, serialAs *asserts.Serial, email string) (string, *osutil.AddUserOptions, error) {
-	errorPrefix := fmt.Sprintf("cannot add system-user %q: ", email)
-
+func createUserWrapper(st *state.State, createData postUserCreateData) ([]devicestate.CreatedUser, error) {
```

Comment:
this is a bit of an atypical naming,  maybe doUserWrapper? or simply remove it and embed it in the caller

Reply:
I'll keep the function, but renamed it to `doUserWrapper`

Reply:
sorry, I typoed my own suggestion, what I meant was doCreateUser, sorry